include/exadg/acoustic_conservation_equations/throughput.h

```diff
@@ -0,0 +1,201 @@
+  throughput.add_parameters(prm);
+
+  try
+  {
+    // we have to assume a default dimension and default Number type
+    // for the automatic generation of a default input file
+    unsigned int const Dim = 2;
+    using Number           = double;
+    Acoustics::get_application<Dim, Number>(input_file, MPI_COMM_WORLD)->add_parameters(prm);
+  }
+  catch(...)
+  {
+  }
+
+  prm.print_parameters(input_file,
+                       dealii::ParameterHandler::Short |
+                         dealii::ParameterHandler::KeepDeclarationOrder);
+}
+
+template<int dim, typename Number>
+void
+run(ThroughputParameters<IncNS::OperatorType> const & throughput,
+    std::string const &                               input_file,
+    unsigned int const                                degree,
+    unsigned int const                                refine_space,
+    unsigned int const                                n_cells_1d,
+    MPI_Comm const &                                  mpi_comm,
+    bool const                                        is_test)
+{
+  std::shared_ptr<IncNS::ApplicationBase<dim, Number>> application =
+    Acoustics::get_application<dim, Number>(input_file, mpi_comm);
+
+  application->set_parameters_throughput_study(degree, refine_space, n_cells_1d);
+
+  std::shared_ptr<Acoustics::Driver<dim, Number>> driver =
+    std::make_shared<Acoustics::Driver<dim, Number>>(mpi_comm, application, is_test, true);
+
+  driver->setup();
+
+  std::tuple<unsigned int, dealii::types::global_dof_index, double> wall_time =
+    driver->apply_operator(throughput.operator_type,
+                           throughput.n_repetitions_inner,
+                           throughput.n_repetitions_outer);
+
+  throughput.wall_times.push_back(wall_time);
+}
+} // namespace ExaDG
+
+int
+main(int argc, char ** argv)
+{
+#ifdef EXADG_WITH_LIKWID
+  LIKWID_MARKER_INIT;
+#endif
+
+  dealii::Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+
+  MPI_Comm mpi_comm(MPI_COMM_WORLD);
+
+  std::string input_file;
+
+  if(argc == 1)
+  {
+    if(dealii::Utilities::MPI::this_mpi_process(mpi_comm) == 0)
+    {
+      // clang-format off
+      std::cout << "To run the program, use:      ./throughput input_file" << std::endl
+                << "To setup the input file, use: ./throughput input_file --help" << std::endl;
+      // clang-format on
+    }
+
+    return 0;
+  }
+  else if(argc >= 2)
+  {
+    input_file = std::string(argv[1]);
+
+    if(argc == 3 and std::string(argv[2]) == "--help")
+    {
+      if(dealii::Utilities::MPI::this_mpi_process(mpi_comm) == 0)
+        ExaDG::create_input_file(input_file);
+
+      return 0;
+    }
+  }
+
+  ExaDG::GeneralParameters                                    general(input_file);
+  ExaDG::HypercubeResolutionParameters                        resolution(input_file, general.dim);
+  ExaDG::ThroughputParameters<ExaDG::Acoustics::OperatorType> throughput(input_file);
+
+  prm.parse_input(input_file, "", true, true);
```

Comment:
For some reasons, I do not see the declaration of this parameter `prm`?

Reply:
I think I forgot to build a "throughput app" when I liked against this version of `ExaDG`. I think there are two possible ways how to proceed here:

1) add a throughput app (no need to review `throughput.h` again).
2) remove `throughput.h` from this PR and introduce it with an app in a follow-up PR.

Probably it's best to go with 2). What do you think?

Reply:
Makes sense.

Reply:
[4625de4](https://github.com/exadg/exadg/pull/592/commits/4625de41a8a9300cb76b1a7ec0161711eb723b7b)